src/main/java/scala_maven/ScalaCompilerSupport.java

```diff
@@ -276,6 +289,104 @@ void setLastSuccessfullTS(long v) throws Exception {
     }
   }
 
+  private ScalaInstance getScalaInstanceForS2(
+      File libraryJar,
+      File reflectJar,
+      File compilerJar,
+      VersionNumber scalaVersion,
+      List<File> extraJars)
+      throws Exception {
+
+    List<File> allJars = new ArrayList<>(extraJars);
+    allJars.add(libraryJar);
+    allJars.add(reflectJar);
+    allJars.add(compilerJar);
+
+    URL[] urls =
+        new URL[] {
+          libraryJar.toURI().toURL(), reflectJar.toURI().toURL(), compilerJar.toURI().toURL()
+        };
+
+    URLClassLoader classLoader = new URLClassLoader(urls);
+    return new ScalaInstance(
+        scalaVersion.toString(), // version
+        classLoader, // loader
+        ClasspathUtil.rootLoader(), // loaderLibraryOnly
+        libraryJar, // libraryJar
+        compilerJar, // compilerJar
+        allJars.toArray(new File[] {}), // allJars
+        Option.apply(scalaVersion.toString()) // explicitActual
+        );
+  }
+
+  private ScalaInstance getScalaInstanceForS3(VersionNumber scalaVersion) throws Exception {
+    String version = scalaVersion.toString();
+    String scalaOrg = getScalaOrganization();
+
+    File[] compilerJars =
+        new File[] {
+          getCompilerJar(scalaVersion),
+          getArtifactJar(scalaOrg, SCALA3_INTERFACES_ARTIFACTID, version),
+          getArtifactJar(scalaOrg, "tasty-core_" + version, version),
+          getArtifactJar("org.scala-lang.modules", "scala-asm", "9.1.0-scala-1"),
+          getArtifactJar("org.scala-sbt", "compiler-interface", "1.3.5"),
+          getArtifactJar("org.jline", "jline-reader", "3.19.0"),
+          getArtifactJar("org.jline", "jline-terminal", "3.19.0"),
+          getArtifactJar("org.jline", "jline-terminal-jna", "3.19.0")
```

Comment:
Rather than fetching all jars individually, it is probably simpler to fetch the compiler and all its dependencies in one call:
```scala
resolver.getJarAndDependencies(scalaOrg, scala3compilerId, version, null).map(_.file).toArray
```

Reply:
then you should be able to filter `scala3-library` and `scala-libary` from compilerJars:

```scala
val libraryJars = compilerJars.filter(jar => jar.getName.contains("scala3-library") || jar.getName.contains("scala-library))
```

Reply:
I address this in my last commit.